polynote-frontend/polynote/cell.js

```diff
@@ -352,21 +353,25 @@ export class CodeCell extends Cell {
         this.updateKeyHandler(KeyPress.of("Enter", {shift: true}).h, (prev) => {
             return () => {
                 this.dispatchEvent(new RunCellEvent(this.id));
-                prev();
+                return prev();
             }
         });
         this.updateKeyHandler(KeyPress.of("Enter", {shift: true, meta: true}).h, (prev) => {
             return () => {
                 this.dispatchEvent(new RunCellEvent(this.id));
-                prev();
+                return prev();
             }
         });
 
         // actually bind keydown
         this.editor.onKeyDown((evt) => {
-            if (this.onKeyDown(evt)) {
-                evt.stopPropagation();
-                evt.preventDefault();
+            // this is really ugly, is there a better way to tell whether the widget is visible??
+            const suggestionsVisible = this.editor.getContribution('editor.contrib.suggestController')._widget._value.suggestWidgetVisible.get();
+            if (!suggestionsVisible) { // don't do stuff when suggestions are visible
+                if (this.onKeyDown(evt)) {
+                    evt.stopPropagation();
+                    evt.preventDefault();
```

Comment:
Using `onKeyDown` instead of `editor.addCommand` means that sometimes we need to prevent a keypress from getting to Monaco if we are overriding it (something that `addCommand` does automatically). So in the case of the new hotkeys for `Shift+Enter` and `Cmd+Shift+Enter`, we need to stop Monaco from creating newlines after the cell focus is switched. To do that we stop propagation/prevent default here. 

Unfortunately this caused the arrow-keys-not-working bug because `this.onKeyDown(evt)` used to return `true` for any match. The way this is solved in this PR is to return `true` only when the handler function wants to prevent propagation, so that the arrow key handlers can allow the movement to go through to monaco while the enter key handler can capture it. So that's what the return value changes above thread through.